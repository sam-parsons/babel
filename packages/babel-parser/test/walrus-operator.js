import { parse } from "../lib";

function getParser(code) {
  return () => parse(code, { sourceType: "module" });
}

describe("walrus operator", function () {
  it("should parse", function () {
    expect(getParser(`if (x := y) alert()`)()).toMatchSnapshot();
  });
});
