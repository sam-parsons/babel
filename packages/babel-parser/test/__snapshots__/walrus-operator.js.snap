// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`walrus operator should parse 1`] = `
Node {
  "comments": Array [],
  "end": 19,
  "errors": Array [],
  "extra": undefined,
  "innerComments": undefined,
  "leadingComments": undefined,
  "loc": SourceLocation {
    "end": Position {
      "column": 19,
      "line": 1,
    },
    "filename": undefined,
    "identifierName": undefined,
    "start": Position {
      "column": 0,
      "line": 1,
    },
  },
  "program": Node {
    "body": Array [
      Node {
        "alternate": null,
        "consequent": Node {
          "end": 19,
          "expression": Node {
            "arguments": Array [],
            "callee": Node {
              "end": 17,
              "extra": undefined,
              "innerComments": undefined,
              "leadingComments": undefined,
              "loc": SourceLocation {
                "end": Position {
                  "column": 17,
                  "line": 1,
                },
                "filename": undefined,
                "identifierName": "alert",
                "start": Position {
                  "column": 12,
                  "line": 1,
                },
              },
              "name": "alert",
              "range": undefined,
              "start": 12,
              "trailingComments": undefined,
              "type": "Identifier",
            },
            "end": 19,
            "extra": undefined,
            "innerComments": undefined,
            "leadingComments": undefined,
            "loc": SourceLocation {
              "end": Position {
                "column": 19,
                "line": 1,
              },
              "filename": undefined,
              "identifierName": undefined,
              "start": Position {
                "column": 12,
                "line": 1,
              },
            },
            "range": undefined,
            "start": 12,
            "trailingComments": undefined,
            "type": "CallExpression",
          },
          "extra": undefined,
          "innerComments": undefined,
          "leadingComments": undefined,
          "loc": SourceLocation {
            "end": Position {
              "column": 19,
              "line": 1,
            },
            "filename": undefined,
            "identifierName": undefined,
            "start": Position {
              "column": 12,
              "line": 1,
            },
          },
          "range": undefined,
          "start": 12,
          "trailingComments": undefined,
          "type": "ExpressionStatement",
        },
        "end": 19,
        "extra": undefined,
        "innerComments": undefined,
        "leadingComments": undefined,
        "loc": SourceLocation {
          "end": Position {
            "column": 19,
            "line": 1,
          },
          "filename": undefined,
          "identifierName": undefined,
          "start": Position {
            "column": 0,
            "line": 1,
          },
        },
        "range": undefined,
        "start": 0,
        "test": Node {
          "end": 10,
          "extra": undefined,
          "innerComments": undefined,
          "leadingComments": undefined,
          "left": Node {
            "end": 5,
            "extra": undefined,
            "innerComments": undefined,
            "leadingComments": undefined,
            "loc": SourceLocation {
              "end": Position {
                "column": 5,
                "line": 1,
              },
              "filename": undefined,
              "identifierName": "x",
              "start": Position {
                "column": 4,
                "line": 1,
              },
            },
            "name": "x",
            "range": undefined,
            "start": 4,
            "trailingComments": undefined,
            "type": "Identifier",
          },
          "loc": SourceLocation {
            "end": Position {
              "column": 10,
              "line": 1,
            },
            "filename": undefined,
            "identifierName": undefined,
            "start": Position {
              "column": 4,
              "line": 1,
            },
          },
          "operator": ":=",
          "range": undefined,
          "right": Node {
            "end": 10,
            "extra": undefined,
            "innerComments": undefined,
            "leadingComments": undefined,
            "loc": SourceLocation {
              "end": Position {
                "column": 10,
                "line": 1,
              },
              "filename": undefined,
              "identifierName": "y",
              "start": Position {
                "column": 9,
                "line": 1,
              },
            },
            "name": "y",
            "range": undefined,
            "start": 9,
            "trailingComments": undefined,
            "type": "Identifier",
          },
          "start": 4,
          "trailingComments": undefined,
          "type": "AssignmentExpression",
        },
        "trailingComments": undefined,
        "type": "IfStatement",
      },
    ],
    "directives": Array [],
    "end": 19,
    "extra": undefined,
    "innerComments": undefined,
    "interpreter": null,
    "leadingComments": undefined,
    "loc": SourceLocation {
      "end": Position {
        "column": 19,
        "line": 1,
      },
      "filename": undefined,
      "identifierName": undefined,
      "start": Position {
        "column": 0,
        "line": 1,
      },
    },
    "range": undefined,
    "sourceType": "module",
    "start": 0,
    "trailingComments": undefined,
    "type": "Program",
  },
  "range": undefined,
  "start": 0,
  "trailingComments": undefined,
  "type": "File",
}
`;
